authors/apps/articles/views.py

```diff
@@ -1,27 +1,46 @@
 from django.shortcuts import render
 from rest_framework import status
-from rest_framework.permissions import IsAuthenticated
+from rest_framework.permissions import IsAuthenticatedOrReadOnly, IsAuthenticated
 from rest_framework.response import Response
 from rest_framework.views import APIView
+from rest_framework.generics import UpdateAPIView
+from django.core import exceptions
 
+from .exceptions import ArticleDoesNotExist
 from .renderers import ArticleJSONRenderer
 from .serializers import CreateArticleSerializer
-
+from .models import Article
 
 class CreateArticleAPIView(APIView):
     permission_classes = (IsAuthenticated,)
+
     renderer_classes = (ArticleJSONRenderer,)
     serializer_class = CreateArticleSerializer
 
     def post(self, request):
         article = request.data.get('article', {})
-        article.update({"author_id": request.user.id})
-
-        serializer = self.serializer_class(data=article)
+        user_data = {"user_data": request.user}
+        serializer = self.serializer_class(data=article, context=user_data)
         serializer.is_valid(raise_exception=True)
-        saved_article = serializer.save()
-        serializer.validated_data.update({"article": saved_article})
+        serializer.save()
+        return Response(serializer.data,status=status.HTTP_201_CREATED)
+
+class UpdateArticleAPIView(UpdateAPIView):
+    queryset = Article.objects.select_related('author')
+    renderer_classes = (ArticleJSONRenderer,)
+    permission_classes = (IsAuthenticatedOrReadOnly,)
```

Comment:
Hi @reifred, this is good work. Keep it up. I have tested the endpoint locally and was able to update an article. The only issues I have found so far is that;
1. Any authenticated user can edit any Article. Kindly restrict this to only the author of the Article.
2. I have not been able to test this new implementation of creating an Article because you have not yet updated this branch. Kindly do so, so that I can re-test.

Overall, this is really good and very insightful, thank you for the work.

Reply:
Thanks @richien, for the feedback. Kindly re-test again with the updated branch now.

Reply:
Hi @reifred, thank you for implementing the feedback. I have tested and the route is working as expected. Great work.